### ISSUE TITLE
Relocate the functions in staging/src/kubevirt.io/client-go/api/utils.go

### ISSUE THREAD

**Issue by zhuchenwang** [kubevirt/kubevirt#7272]
This is a request for internal cleanup.

The  `staging/src/kubevirt.io/client-go/api/utils.go` only contains 2 util functions used for test only.

Our code base depends on the `kubevirt.io/kubevirt/tests` package. Also, we have a higher `kube-openapi` version. Now the `kube-openapi` version conflicts with the version that is used in Kubevirt. This is because the https://github.com/kubevirt/kubevirt/blob/main/tests/utils.go file imports the `kubevirt.io/client-go/api` only to use the util functions in https://github.com/kubevirt/kubevirt/blob/main/staging/src/kubevirt.io/client-go/api/utils.go. 

Also, by looking at the util functions in https://github.com/kubevirt/kubevirt/blob/main/staging/src/kubevirt.io/client-go/api/utils.go, I feel they are more aligned with the functions in https://github.com/kubevirt/kubevirt/blob/main/tests/utils.go.

I suggest to just move the functions from https://github.com/kubevirt/kubevirt/blob/main/staging/src/kubevirt.io/client-go/api/utils.go to https://github.com/kubevirt/kubevirt/blob/main/tests/utils.go, so that depending on `kubevirt.io/kubevirt/tests` won't bring the `kube-openapi` dependency transitively.

I am happy to send a PR. Please let me know if there are any concerns.

**Comment by zhuchenwang** [comment]
Hi there, 

Are there any thoughts regarding to the proposal?

**Comment by xpivarc** [comment]
Hi, I think it makes sense. Feel free to open PR.

**Comment by zhuchenwang** [comment]
I just take a deep look at the code. The test functions are also used in `kubevirt/pkg/..`. Simply move them to the `kubevirt/tests/utils.go` will cause circular dependencies.

However, I still want to decouple it from the openapi generated code. I think it would be better to move the [staging/src/kubevirt.io/client-go/api/utils.go](https://github.com/kubevirt/kubevirt/blob/main/staging/src/kubevirt.io/client-go/api/utils.go) file to `staging/src/kubevirt.io/api/utils.go`.

**Comment by rmohr** [comment]
@zhuchenwang I did not get to untagle this function at the time when I started with our clean `api` package. How about leaving the function in `client-go`? It really is only there for kubevirt-internal testing and has no real meaning for others.

There are just two usages in the `/tests` package. I think it is worth just "duplicating" a proper function for `/tests`. We can then independently evolve the helper for unit-tests and e2e tests and remove a dependency which does not really make sense.
